app/controllers/devise/two_factor_authentication_controller.rb

```diff
@@ -12,19 +13,32 @@ class TwoFactorAuthenticationController < DeviseController
     def new
       analytics.track_event('User requested a new OTP code')
 
-      current_user.send_new_otp
+      send_user_otp
+
+      show_direct_otp_prompt
       flash[:notice] = t('devise.two_factor_authentication.user.new_otp_sent')
-      redirect_to user_two_factor_authentication_path(method: 'sms')
     end
 
     def show
       if use_totp?
         show_totp_prompt
       else
-        show_direct_otp_prompt
+        @phone_number = user_decorator.masked_two_factor_phone_number
       end
     end
 
+    def confirm
+      @phone_number = user_decorator.masked_two_factor_phone_number
+      @code_value = current_user.direct_otp if FeatureManagement.prefill_otp_codes?
+    end
```

Comment:
This method seems to be improperly named. We've been using `confirm` to process a POST request containing the code entered by the user so we can determine whether or not the code is valid. Here, it's being used for a GET request to set up view variables. Why this change?